### DISCUSSION TITLE
Use gc service account for Kafka Connect output build

### DISCUSSION THREAD

**Discussion by holysheep** [strimzi/strimzi-kafka-operator#7775] [Q&A] [answered]
I want to push Kafka's Connect build output image with debezium connector to the Google Cloud private registry. I prefer to use service account instead of secret for this.

So I've added annotation in KafkaConnect's template for the SA patching like this:

<img width="550" alt="Screenshot 2022-12-09 at 19 04 40" src="https://user-images.githubusercontent.com/9785806/206765928-e5db7a97-6fc3-4bb6-89ea-81e59b1f1fd3.png">

But it does not annotate `strimzi-kafka-connect-cluster-connect-build` service account, which is addressing docker registry:

<img width="796" alt="Screenshot 2022-12-09 at 19 05 00" src="https://user-images.githubusercontent.com/9785806/206765757-b3fa2f64-0863-4bbe-995f-402530b4468e.png">

And `strimzi-kafka-connect-cluster-connect` doesnt get annotated either.

Am I doing something wrong or is it not possible to push output artifact without a pushSecret?

**Comment by scholzj** [comment, marked as answer]
If you want to assign an annotation to the builder pod, you have to do it in [`.spec.template.buildPod`](https://strimzi.io/docs/operators/latest/full/configuring.html#type-KafkaConnectTemplate-reference) and not in `.spec.template.pod`. `.spec.template.pod` would set the annotation to the Connect worker pods.

But none of that would annotate the Service Account. If you want to annotate the Service Account used for the build, it would be `.spec.template.buildServiceAccount` ;-)
